GliaWidgets/SecureConversations/FileUploadListView/SecureConversations.FilePreviewView.swift

```diff
@@ -0,0 +1,232 @@
+#if canImport(QuickLookThumbnailing)
+    import QuickLookThumbnailing
+#endif
+
+import UIKit
+
+extension SecureConversations {
+    class FilePreviewView: UIView {
+        enum Kind: Equatable {
+            case none
+            case file(LocalFile)
+            case fileExtension(String?)
+            case error
+        }
+
+        struct Props: Equatable {
+            let style: FilePreviewStyle
+            let kind: Kind
+        }
+
+        var props: Props {
+            didSet {
+                renderProps()
+            }
+        }
+
+        private let imageView = UIImageView()
+        private let label = UILabel()
+        private let kSize = CGSize(width: 52, height: 52)
+
+        init() {
+            self.props = Props(style: .initial, kind: .none)
+            super.init(frame: .zero)
+            setup()
+            layout()
+            renderProps()
+        }
+
+        @available(*, unavailable)
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        private func setup() {
+            clipsToBounds = true
+            label.textAlignment = .center
+        }
+
+        private func layout() {
+            NSLayoutConstraint.autoSetPriority(.defaultHigh) {
+                autoSetDimension(.height, toSize: kSize.height)
+            }
+            autoSetDimension(.width, toSize: kSize.width)
+
+            addSubview(imageView)
+            imageView.autoPinEdgesToSuperviewEdges()
+
+            addSubview(label)
+            label.autoPinEdge(toSuperviewEdge: .left, withInset: 5)
+            label.autoPinEdge(toSuperviewEdge: .right, withInset: 5)
+            label.autoAlignAxis(toSuperviewAxis: .horizontal)
+        }
+
+        func renderProps() {
```

Comment:
Can it be `private`?

Reply:
Added here 26bd52271dfcf47c510b4bbae624baa646c8fe43.